packages/react-router/modules/matchPath.js

```diff
@@ -26,11 +26,17 @@ const compilePath = (pattern, options) => {
 /**
  * Public API for matching a URL pathname to a path pattern.
  */
-const matchPath = (pathname, path, options = {}) => {
+const matchPath = (pathname, path, options = {}, parentMatch = null) => {
   const { exact = false, strict = false } = options
 
-  if (!path)
-    return { url: pathname, isExact: true, params: {} }
+  const absolute = isAbsolute(path)
+
+  if (path === undefined)
+    return { url: parentMatch ? parentMatch.url : '/', isExact: true, params: {} }
```

Comment:
This used to be that when `path` was falsy, it would return a match with `url: pathname`. This broke resolving because when a relative child `<Route>` would be resolved, `parentMatch.url` would be the full URL, not just the previously matched part of the URL.

If the user needs the full pathname, they can always access it through `location.pathname`.